spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/outbound/MongoDbOutboundGateway.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mongodb.outbound;
+
+import java.util.List;
+
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.convert.MongoConverter;
+import org.springframework.data.mongodb.core.query.BasicQuery;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.TypeLocator;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.expression.spel.support.StandardTypeLocator;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.handler.AbstractReplyProducingMessageHandler;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+
+import com.mongodb.DBObject;
+
+/**
+ * Makes outbound operations to query a MongoDb database using a {@link MongoOperations}
+ *
+ * @author Xavier Padró
+ * @since 5.0
+ */
+public class MongoDbOutboundGateway extends AbstractReplyProducingMessageHandler {
+
+	private volatile boolean initialized = false;
+
+	private static final String DEFAULT_COLLECTION_NAME = "data";
+
+	private volatile EvaluationContext evaluationContext;
+
+	private volatile String mongoDbQuery;
+
+	private volatile Expression mongoDbQueryExpression;
+
+	private volatile MongoOperations mongoTemplate;
+
+	private volatile MongoDbFactory mongoDbFactory;
+
+	private volatile MongoConverter mongoConverter;
+
+	private volatile boolean expectSingleResult = false;
+
+	private volatile Class<?> entityClass = DBObject.class;
```

Comment:
I can't right now "why?" but looks like since the move to the MongoDB 3.x it would be better to make default to the `org.bson.Document` . Fix the same in the XSD attribute description.